src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -723,6 +723,11 @@ public static bool CheckConstraints(this NamedTypeSymbol type, in CheckConstrain
         // metadata and be instantiated in C#.  We check to see if that's happened.
         private static bool HasDuplicateInterfaces(NamedTypeSymbol type, ConsList<TypeSymbol> basesBeingResolved)
         {
+            if (type.OriginalDefinition is SourceNamedTypeSymbol)
```

Comment:
>if (type.OriginalDefinition is SourceNamedTypeSymbol) [](http://example.com/codeflow?start=12&length=53)

It doesn't look like this is going to address the problem for a retargeting scenario. Please add a test like that.

Reply:
If the check is purely about types coming from metadata, then that is what we should be checking for instead?